python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -250,13 +250,39 @@ keys:
           convert_types:
             - str
           description: VLAN tag to configure on sub-interface.
+          deprecation:
+            warning: true
+            remove_in_version: 6.0.0
+            new_key: encapsulation_dot1q.vlan
+        encapsulation_dot1q:
+          description: |-
+            Warning: `encapsulation_dot1q` should not be combined with `ethernet_interfaces[].type == l3dot1q/l2dot1q`.
+          type: dict
+          keys:
+            vlan:
+              type: int
+              convert_types:
+                - str
+              required: true
+              description: VLAD ID.
+            inner_vlan:
+              type: int
+              convert_types:
+                - str
+              description: Inner VLAN ID. This setting can only be applied to sub-interfaces on EOS.
         encapsulation_vlan:
           type: dict
+          description: |-
+            This setting can only be applied to sub-interfaces on EOS.
+            Warning: `encapsulation_vlan` should not be combined with `ethernet_interfaces[].type == l3dot1q/l2dot1q`.
           keys:
             client:
               type: dict
               keys:
                 dot1q:
+                  deprecation:
```

Comment:
As the parameters for other protocols are same and only one protocol can be applied at a time, so instead of adding a separate same block for `dot1ad` in the schema and repeated if/else conditions in template, Claus and myself came up with this schema having valid-values for protocol.